### PR TITLE
Add title to pages using page_title helper

### PR DESCRIPTION
**Pull request by chubberlisk**
### Context

Currently all pages have the same HTML title: "Apply for postgraduate teacher training - GOV.UK". This should be dynamic and depend on individual pages.

### Changes proposed in this pull request

A `page_title` helper is used to create the title using what is defined in `config/locales/en.yml` and in each view it is called to set `:title`.

### Guidance to review

It would be good to get feedback on the way this has been done and particularly the tests.

### Link to Trello card

[712 - Set correct page titles for all the pages](https://trello.com/c/Nlc67KGJ/712-set-correct-page-titles-for-all-the-pages)

### CODE REVIEW

**Comment by duncanjbrown** [comment]
Not a formal review: a) this looks good, b) why have two examples covering the same spec?

**Comment by chubberlisk** [comment]
> Not a formal review: a) this looks good, b) why have two examples covering the same spec?

So the two examples are the result of triangulation when TDDing. It ensures that the code is generic and not hardcorded to one solution.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,2 +1,11 @@
 module ApplicationHelper
+  def page_title(page:)
+    page_title_translation_key = "page_titles.#{page}"
+
+    if I18n.exists?(page_title_translation_key, :en)
+      "#{t(page_title_translation_key)} - #{t('page_titles.application')}"
+    else
+      t('page_titles.application')
+    end
+  end
```
Comment (alexdesi) [inline app/helpers/application_helper.rb]: You might want to replace the `if`, something like:  (up to you)
```
title = t('page_titles.application').present? ? t('page_titles.application') :  ''
t(page_title_translation_key) + title
```

File: spec/helpers/application_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+describe ApplicationHelper do
+  describe '#page_title' do
+    context 'given a page that does not exist' do
```
Comment (alexdesi) [inline spec/helpers/application_helper_spec.rb]: Actually is not the page that does not exist, is the `page name`

File: spec/helpers/application_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      it 'returns the application title' do
+        page_title = helper.page_title(page: :meow)
```
Comment (alexdesi) [inline spec/helpers/application_helper_spec.rb]: do we need a hash params for the page, it is just a one param, what about
`page_title_for(:meow)`

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (chubberlisk) [reply app/helpers/application_helper.rb]: I've decided to keep it how it is as I like that it's really simple and easy to read.

File: spec/helpers/application_helper_spec.rb (hunk 3 above)
Comment (chubberlisk) [reply spec/helpers/application_helper_spec.rb]: Makes sense!

File: spec/helpers/application_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        page_title = helper.page_title(page: :personal_details)
+
+        expect(page_title).to eq("#{t('page_titles.personal_details')} - #{t('page_titles.application')}")
+      end
+
+      it 'returns the page name with the application title (example two)' do
+        page_title = helper.page_title(page: :contact_details)
+
+        expect(page_title).to eq("#{t('page_titles.contact_details')} - #{t('page_titles.application')}")
+      end
```
[the remark this answers is not in the document]
Comment (chubberlisk) [reply spec/helpers/application_helper_spec.rb]: We had a discussion IRL and decided that it would be best to remove the second test as it's now redundant and not an edge case test.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (alexdesi) [reply app/helpers/application_helper.rb]: cool, thanks for removing `:en`

File: spec/helpers/application_helper_spec.rb (hunk 2 above)
Comment (chubberlisk) [reply spec/helpers/application_helper_spec.rb]: Yes, you're right! I've updated the context description to be clearer, I changed it to:

- `given a page is not defined in the translation file` and
- `given a page is defined in the translation file`

Hopefully this is clearer?
